src/Dofs/apply_analytical.jl

```diff
@@ -0,0 +1,114 @@
+
+!!! note
+    This function only works for standard nodal finite element interpolations 
+    when the function value at the (algebraic) node is equal to the corresponding 
+    degree of freedom value.
+    This holds for e.g. Lagrange and Serendipity interpolations, including 
+    sub- and superparametric elements. 
+
+"""
+function apply_analytical!(
+    a::AbstractVector, dh::DofHandler, f::Function, 
+    fieldname::Union{Symbol,Nothing}=nothing,
+    cellset=1:getncells(dh.grid)
+    )
+    field = _check_fieldname(dh, fieldname)
+    ip_geo=_default_interpolation(dh)
+    field_idx = find_field(dh, field)
+    ip_fun = getfieldinterpolation(dh, field_idx)
+    celldofinds = dof_range(dh, field)
+    field_dim = getfielddim(dh, field_idx)
+    _apply_analytical!(a, dh, celldofinds, field_dim, ip_fun, ip_geo, f, cellset)
+end
+
+function apply_analytical!(
+    a::AbstractVector, dh::MixedDofHandler, f::Function, 
+    fieldname::Union{Symbol,Nothing}=nothing,
+    cellset=1:getncells(dh.grid),
+    )
+    field = _check_fieldname(dh, fieldname)
+    ip_geos=_default_interpolations(dh)
+
+    for (fh, ip_geo) in zip(dh.fieldhandlers, ip_geos)
+        field ∈ getfieldnames(fh) || continue
+        field_idx = find_field(fh, field)
+        ip_fun = getfieldinterpolation(fh, field_idx)
+        field_dim = getfielddim(fh, field_idx)
+        celldofinds = dof_range(fh, field)
+        _apply_analytical!(a, dh, celldofinds, field_dim, ip_fun, ip_geo, f, intersect(fh.cellset, cellset))
+    end
+    return a
+end
+
+function _apply_analytical!(
+    a::Vector, dh::AbstractDofHandler, celldofinds, field_dim,
+    ip_fun::Interpolation, ip_geo::Interpolation, f::Function, cellset)
+    
+    coords = getcoordinates(dh.grid, first(cellset))
+    c_dofs = celldofs(dh, first(cellset))
+    f_dofs = zeros(Int, length(celldofinds))
+
+    # Check f before looping
+    length(f(first(coords))) == field_dim || error("length(f(x)) must be equal to dimension of the field ($field_dim)")
+
+    for cellnr in cellset
+        cellcoords!(coords, dh, cellnr)
+        celldofs!(c_dofs, dh, cellnr)
+        # f_dofs .= c_dofs[celldofinds]
+        foreach(i->(f_dofs[i] = c_dofs[celldofinds[i]]), 1:length(celldofinds)) 
+        _apply_analytical!(a, f_dofs, coords, field_dim, ip_fun, ip_geo, f)
+    end
+    return a
+end
+
+function _apply_analytical!(a::Vector, dofs::Vector{Int}, coords::Vector{XT}, field_dim, ip_fun, ip_geo, f) where {XT<:Vec}
+
+    getnbasefunctions(ip_geo) == length(coords) || error("coords=$coords not compatible with ip_ge=$ip_geo")
+    ref_coords = reference_coordinates(ip_fun)
+    length(ref_coords)*field_dim == length(dofs) || error("$ip_fun must have length(dofs)=$(length(dofs)) reference coords")
+
+    for (i_dof, refpoint) in enumerate(ref_coords)
+        x_dof = zero(XT)
+        for (i,x_node) in enumerate(coords)
+            x_dof += value(ip_geo, i, refpoint) * x_node
+        end
```

Comment:
Feels like this exist somewhere, maybe `BCValues`?

Reply:
Yes,
https://github.com/Ferrite-FEM/Ferrite.jl/blob/2f8655970f29a7d37d41a5e721cdf10d8b21ddbc/src/FEValues/face_values.jl#L264
and 
https://github.com/Ferrite-FEM/Ferrite.jl/blob/2f8655970f29a7d37d41a5e721cdf10d8b21ddbc/src/FEValues/face_values.jl#L278
but this seems specialized for faces? In that case, we would call `f(x)` even more times?
(I can create a similar construct pre-caching the geometric shape values at the reference points if desired, but I feel the code will become complicated then. )

Reply:
Introduced `CellDofValues` now. 
I'm not sure about the name (as usual), but it is internal.